lib/ace/ext/easymotion.js

```diff
@@ -0,0 +1,243 @@
+    }
+}
+
+var accuInput = "";
+function easymotionJump(editor, char) {
+    accuInput = accuInput + char;
+    var idx = accuInput;
+    var keys = Object.keys(findings);
+    var matchingKeys = [];
+
+    var i;
+    for (i = 0; i < keys.length; i++) {
+      var k = keys[i];
+      if (k.slice(0, accuInput.length) === accuInput) {
+        matchingKeys.push(k);
+      }
+    }
+
+    if (matchingKeys.length === 0)
+      return true;
+    if (findings[idx] && matchingKeys.length === 1) {
+      editor.moveCursorTo(findings[idx].row, findings[idx].column);
+      editor.selection.clearSelection();
+      return true;
+    }
+
+    updateMarkers(editor);
+
+    return false;
+}
+
+function updateMarkers(editor) {
+    var keys = Object.keys(findings);
+    var i;
+    for (i = 0; i < keys.length; i++) {
+        var k = keys[i];
+        var v = findings[k];
+
+        if (keys[i].slice(0, accuInput.length) !== accuInput) {
+          editor.session.removeMarker(v.markerId);
+        }
+    }
+}
+
+function clearMarkers(editor) {
+    var keys = Object.keys(findings);
+    var i;
+    for (i = 0; i < keys.length; i++) {
+        var k = keys[i];
+        var v = findings[k];
+
+        editor.session.removeMarker(v.markerId);
+    }
+}
+
+var easymotionPhase = 0;
+
+function easymotionHandler(data, hashId, key, keyCode) {
+    var editor = data.editor;
+    if (keyCode == -1) return;
+    if (key.length > 1) {
+      if (key == "space") key = " ";
+      if (key[0] == "n") key = key.replace("numpad", "");
+    }
+    if ((hashId === 0 || hashId === 4) && key.length == 1)
+        return {command: "null", passEvent: true}; // wait until input event
+    if (hashId == -1)  {
+      // easymotion functionality in two phases
+      if (easymotionPhase === 0) {
+        easymotionMark(editor, key);
+        easymotionPhase++;
+        accuInput = "";
+      } else if (easymotionPhase === 1) {
+        var ret = easymotionJump(editor, key);
+        if (ret) {
+          easymotionHandlerCleanup({editor: editor});
+          easymotionPhase = 0;
+        }
+      }
+      return {command: "null"}
+    } else {
+        // some other key: exit easymotion mode
+        easymotionHandlerCleanup({editor: editor});
+        clearMarkers(editor);
+        easymotionPhase = 0;
+        return {command: "null"}
+    }
+};
+
+function easymotionHandlerSetup(editor) {
+    editor.keyBinding.addKeyboardHandler(easymotionHandler);
+    editor.on("mousedown", easymotionHandlerCleanup);
+}
+
+function easymotionHandlerCleanup(event) {
+    var editor = event.editor;
+    editor.keyBinding.removeKeyboardHandler(easymotionHandler)
+    editor.off("mousedown", easymotionHandlerCleanup)
+
+    clearMarkers(editor);
+}
+
+exports.easymotionSetup = function(editor) {
+    easymotionHandlerSetup(editor);
+};
+
+exports.commands = [{
+    name: "easymotion-setup",
+    exec: function(editor) {
+        exports.easymotionSetup(editor);
+    },
+    bindKey: "Ctrl-Super+F"
```

Comment:
Super-F is not available on windows, Using `{mac "Ctrl-Cmd-F", win: "alt-F"}` might be better.
Which keybindings other easymotion plugins use?